storage/metricsstore/metrics/decorator.go

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) 2022 The Jaeger Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package metrics
+
+import (
+	"context"
+	"time"
+
+	"github.com/uber/jaeger-lib/metrics"
+
+	protometrics "github.com/jaegertracing/jaeger/proto-gen/api_v2/metrics"
+	"github.com/jaegertracing/jaeger/storage/metricsstore"
+)
+
+// ReadMetricsDecorator wraps a metricsstore.Reader and collects metrics around each read operation.
+type ReadMetricsDecorator struct {
+	reader                    metricsstore.Reader
+	getLatenciesMetrics       *queryMetrics
+	getCallRatesMetrics       *queryMetrics
+	getErrorRatesMetrics      *queryMetrics
+	getMinStepDurationMetrics *queryMetrics
+}
+
+type queryMetrics struct {
+	Errors     metrics.Counter `metric:"requests" tags:"result=err"`
+	Successes  metrics.Counter `metric:"requests" tags:"result=ok"`
+	Responses  metrics.Timer   `metric:"responses"` // used as a histogram, not necessary for GetTrace
+	ErrLatency metrics.Timer   `metric:"latency" tags:"result=err"`
+	OKLatency  metrics.Timer   `metric:"latency" tags:"result=ok"`
+}
+
+func (q *queryMetrics) emit(err error, latency time.Duration, responses int) {
+	if err != nil {
+		q.Errors.Inc(1)
+		q.ErrLatency.Record(latency)
+	} else {
+		q.Successes.Inc(1)
+		q.OKLatency.Record(latency)
+		q.Responses.Record(time.Duration(responses))
```

Comment:
I don't follow, what is 1ns here?

Reply:
Good question, I actually don't really understand how it was used in [spanstore](https://github.com/jaegertracing/jaeger/blob/bf64373d1e690594fd8c279720faf32722cf1494/storage/spanstore/metrics/decorator.go#L53) either. 

Either way, I don't think it applies for the metrics reader use case so I've removed it in [857f64a](https://github.com/jaegertracing/jaeger/pull/3667/commits/857f64a8c066fa0ec8c8d7532c56212122da35a4).